### PR TITLE
HBASE-23834. HBase fails to run on Hadoop 3.3.0/3.2.2/3.1.4 due to jetty version mismatch.

### PR DESCRIPTION
**Pull request by jojochuang**
(1) updated Jetty version.
(2) hadoop-3 profile uses Hadoop 3.1.4-SNAPSHOT which runs on Jetty 9.4.
There's not a Hadoop 2 version that runs on Jetty 9.4 so skip it for now.

Create this PR and let it run through the precommit test.

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 28s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 28s |  master passed  |
| +1 :green_heart: |  compile  |   3m 19s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m  9s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   3m  8s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 58s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 30s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 30s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  shadedjars  |   5m 15s |  patch has no errors when building our shaded downstream artifacts.  |
| -1 :x: |  hadoopcheck  |  15m 11s |  The patch causes 52 errors with Hadoop v3.1.2.  |
| +1 :green_heart: |  javadoc  |   2m 57s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   5m 55s |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 17s |  The patch does not generate ASF License warnings.  |
|  |   |  59m 47s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.http.TestHttpServer |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1180/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1180 |
| Optional Tests | dupname asflicense javac javadoc unit shadedjars hadoopcheck xml compile |
| uname | Linux 13b5584dbadb 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1180/out/precommit/personality/provided.sh |
| git revision | master / e11dbf36a5 |
| Default Java | 1.8.0_181 |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1180/1/artifact/out/patch-javac-3.1.2.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1180/1/artifact/out/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1180/1/testReport/ |
| Max. process+thread count | 470 (vs. ulimit of 10000) |
| modules | C: . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1180/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by jojochuang** [comment]
Not sure if it's caused by my change. Triggered a rebuild of the precommit

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 39s |  master passed  |
| +1 :green_heart: |  compile  |   3m  3s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 39s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   2m 56s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 59s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m  2s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m  2s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  shadedjars  |   4m 37s |  patch has no errors when building our shaded downstream artifacts.  |
| -1 :x: |  hadoopcheck  |  12m 39s |  The patch causes 52 errors with Hadoop v3.1.2.  |
| +1 :green_heart: |  javadoc  |   2m 55s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   5m  4s |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 17s |  The patch does not generate ASF License warnings.  |
|  |   |  51m 26s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.http.TestHttpServer |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.6 Server=19.03.6 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1180/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1180 |
| Optional Tests | dupname asflicense javac javadoc unit shadedjars hadoopcheck xml compile |
| uname | Linux e4cd8897e4ff 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1180/out/precommit/personality/provided.sh |
| git revision | master / 86e76c2bd9 |
| Default Java | 1.8.0_181 |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1180/2/artifact/out/patch-javac-3.1.2.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1180/2/artifact/out/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1180/2/testReport/ |
| Max. process+thread count | 665 (vs. ulimit of 10000) |
| modules | C: . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1180/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: pom.xml [hunk 1]
```diff
@@ -1471,7 +1471,7 @@
     <java.min.version>${compileSource}</java.min.version>
     <!-- Dependencies -->
     <hadoop-two.version>2.8.5</hadoop-two.version>
-    <hadoop-three.version>3.1.2</hadoop-three.version>
+    <hadoop-three.version>3.1.4-SNAPSHOT</hadoop-three.version>
```
Comment (busbey) [inline pom.xml]: no depending on snapshot versions in the committed codebase. we don't include references to snapshot repositories, so we shouldn't be able to find any.

**Pull request closed without merging**